### ISSUE TITLE
Float and Double comparing bug

### ISSUE THREAD

**Issue by rodion-m** [ncalc/ncalc#29]
The code below:
```csharp
            var e = new Expression("f == 0.33");
            e.Parameters["f"] = 0.33f;
            Assert.AreEqual(true, e.Evaluate());
```
Fails. I assume it's a bug.
Here is the source of the problem:
![image](https://user-images.githubusercontent.com/36400912/129356388-f3695f61-19ae-4eb0-8dbe-0eb4e62c3aa4.png)

**Comment by yallie** [comment]
I wouldn't say it's a bug because it's a known issue of all binary floating point data types.
Finite decimal floating point values aren't always represented as finite binary values.
So if you try comparing the same numbers in plain C#, you'll get the same results:

```
using System;

class Program
{
	static void Main()
	{
		float f33 = 0.33f;
		double d33 = 0.33;

		Console.WriteLine(f33 == d33);
	}
}
```

The program above prints `False`.

Anyway, I agree that your proposed evaluation option is a good thing to have 👍
Thanks for your pull request!

**Comment by gumbarros** [comment]
Solved by #117
